src/application/extension-popup-menu/app.tsx

```diff
@@ -0,0 +1,55 @@
+import { Route, Routes } from 'react-router-dom';
+
+import { useExtensionSettings } from 'shared/extension';
+import { Closable } from 'shared/ui';
+
+import { Footer } from './footer';
+import { PopupHomeView } from './popup-home-view';
+import { TopBar } from './top-bar';
+import { CustomizationSettings } from './settings-view/customization-settings';
+import { MainSettingsMenu } from './settings-view/main-settings-menu';
+
+export const EXTENSION_POPUP_MENU_ROUTES = {
+  HOME: '/',
+  SETTINGS: {
+    MAIN_VIEW: '/settings',
+    CUSTOMIZATION: '/settings/customization',
+  },
+};
+
+export const App = () => {
+  const { isPopupMenuVisible, hidePopupMenu } = useExtensionSettings();
```

Comment:
imho extension settings and popup should be decoupled, the fact that popup holds settings inside of it is just a detail that should not be important for consumers of popup, so I would recommend to do:
```ts
const { hide, open, isVisible } = useExtensionPopup();
```

it also looks nice when you're not destructuring:
```ts
const extensionPopup = useExtensionPopup();
extensionPopup.open(EXTENSION_POPUP_ROUTE.SETTINGS_CUSTOMIZATION);
```

Reply:
Great idea, looks much better 👌